Codegen issue: integer vs double confusion causing compilation failure

## Hello!
<!-- Please leave this section as-is, it's designed to help others in the community know how to interact with our GitHub issues. -->

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

This came up when trying to compile poulumi-azure-native, for example:

```
/Users/anton/pulumi-java/providers/pulumi-azure-native/sdk/java/src/main/java/io/pulumi/azurenative/netapp/VolumeArgs.java:248: error: incompatible types: bad type in conditional expression
        this.throughputMibps = throughputMibps == null ? Input.ofNullable(0) : throughputMibps;
                                                                         ^
    inference variable T has incompatible bounds
      equality constraints: Double
      lower bounds: Integer
  where T is a type-variable:
    T extends Object declared in method <T>ofNullable(T)
```


<!-- Please provide a general summary of the issue, including what you're trying to accomplish. -->

### Steps to reproduce
<!-- Provide a link to a live example or an unambiguous set of steps to reproduce this bug. Tell us what you expected to happen, and what happened instead. Please include code to reproduce, if relevant. -->

1.
2.
3.
4.

Expected: <!-- What did you expect to happen? -->
Actual: <!-- What happens instead? -->

--- a/pkg/codegen/jvm/gen.go
+++ b/pkg/codegen/jvm/gen.go
@@ -954,7 +954,7 @@ func primitiveValue(value interface{}) (string, string, error) {
 	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32:
 		return strconv.FormatUint(v.Uint(), 10), "integer", nil
 	case reflect.Float32, reflect.Float64:
-		return strconv.FormatFloat(v.Float(), 'G', -1, 64), "number", nil
+		return strconv.FormatFloat(v.Float(), 'e', -1, 64), "number", nil
 	case reflect.String:
 		return fmt.Sprintf("%q", v.String()), "string", nil
 	default:

--- a/pkg/codegen/testing/test/testdata/mini-azurenative/jvm/src/main/java/io/pulumi/azurenative/netapp/Volume.java
+++ b/pkg/codegen/testing/test/testdata/mini-azurenative/jvm/src/main/java/io/pulumi/azurenative/netapp/Volume.java
@@ -14,6 +14,12 @@ import javax.annotation.Nullable;
 
 @ResourceType(type="azure-native:netapp:Volume")
 public class Volume extends io.pulumi.resources.CustomResource {
+    @OutputExport(name="throughputMibps", type=Double.class, parameters={})
+    private Output</* @Nullable */ Double> throughputMibps;
+
+    public Output</* @Nullable */ Double> getThroughputMibps() {
+        return this.throughputMibps;
+    }
     @OutputExport(name="usageThreshold", type=Double.class, parameters={})
     private Output<Double> usageThreshold;
 

--- a/pkg/codegen/testing/test/testdata/mini-azurenative/jvm/src/main/java/io/pulumi/azurenative/netapp/VolumeArgs.java
+++ b/pkg/codegen/testing/test/testdata/mini-azurenative/jvm/src/main/java/io/pulumi/azurenative/netapp/VolumeArgs.java
@@ -7,11 +7,19 @@ import io.pulumi.core.Input;
 import io.pulumi.core.internal.annotations.InputImport;
 import java.lang.Double;
 import java.util.Objects;
+import javax.annotation.Nullable;
 
 
 public final class VolumeArgs extends io.pulumi.resources.ResourceArgs {
 
     public static final VolumeArgs Empty = new VolumeArgs();
+
+    @InputImport(name="throughputMibps")
+    private final @Nullable Input<Double> throughputMibps;
+
+    public Input<Double> getThroughputMibps() {
+        return this.throughputMibps == null ? Input.empty() : this.throughputMibps;
+    }
 
     @InputImport(name="usageThreshold", required=true)
     private final Input<Double> usageThreshold;
@@ -20,11 +28,15 @@ public final class VolumeArgs extends io.pulumi.resources.ResourceArgs {
         return this.usageThreshold;
     }
 
-    public VolumeArgs(Input<Double> usageThreshold) {
-        this.usageThreshold = usageThreshold == null ? Input.ofNullable(1.073741824E+11) : Objects.requireNonNull(usageThreshold, "expected parameter 'usageThreshold' to be non-null");
+    public VolumeArgs(
+        @Nullable Input<Double> throughputMibps,
+        Input<Double> usageThreshold) {
+        this.throughputMibps = throughputMibps == null ? Input.ofNullable(0e+00) : throughputMibps;
+        this.usageThreshold = usageThreshold == null ? Input.ofNullable(1.073741824e+11) : Objects.requireNonNull(usageThreshold, "expected parameter 'usageThreshold' to be non-null");
     }
 
     private VolumeArgs() {
+        this.throughputMibps = Input.empty();
         this.usageThreshold = Input.empty();
     }
 
@@ -37,6 +49,7 @@ public final class VolumeArgs extends io.pulumi.resources.ResourceArgs {
     }
 
     public static final class Builder {
+        private @Nullable Input<Double> throughputMibps;
         private Input<Double> usageThreshold;
 
         public Builder() {
@@ -45,7 +58,18 @@ public final class VolumeArgs extends io.pulumi.resources.ResourceArgs {
 
         public Builder(VolumeArgs defaults) {
     	      Objects.requireNonNull(defaults);
+    	      this.throughputMibps = defaults.throughputMibps;
     	      this.usageThreshold = defaults.usageThreshold;
+        }
+
+        public Builder setThroughputMibps(@Nullable Input<Double> throughputMibps) {
+            this.throughputMibps = throughputMibps;
+            return this;
+        }
+
+        public Builder setThroughputMibps(@Nullable Double throughputMibps) {
+            this.throughputMibps = Input.ofNullable(throughputMibps);
+            return this;
         }
 
         public Builder setUsageThreshold(Input<Double> usageThreshold) {
@@ -59,7 +83,7 @@ public final class VolumeArgs extends io.pulumi.resources.ResourceArgs {
         }
 
         public VolumeArgs build() {
-            return new VolumeArgs(usageThreshold);
+            return new VolumeArgs(throughputMibps, usageThreshold);
         }
     }
 }

--- a/pkg/codegen/testing/test/testdata/mini-azurenative/schema.json
+++ b/pkg/codegen/testing/test/testdata/mini-azurenative/schema.json
@@ -8,6 +8,10 @@
           "type": "number",
           "description": "Maximum storage quota allowed for a file system in bytes. This is a soft quota used for alerting only. Minimum size is 100 GiB. Upper limit is 100TiB. Specified in bytes.",
           "default": 107374182400
+        },
+        "throughputMibps": {
+          "type": "number",
+          "default": 0
         }
       },
       "type": "object",
@@ -19,6 +23,10 @@
           "type": "number",
           "description": "Maximum storage quota allowed for a file system in bytes. This is a soft quota used for alerting only. Minimum size is 100 GiB. Upper limit is 100TiB. Specified in bytes.",
           "default": 107374182400
+        },
+        "throughputMibps": {
+          "type": "number",
+          "default": 0
         }
       },
       "requiredInputs": [
@@ -409,15 +417,15 @@
         ]
       },
       "outputs": {
-          "description": "A web app, a mobile app backend, or an API app.",
-          "properties": {
-              "siteConfig": {
-                  "type": "object",
-                  "$ref": "#/types/azure-native:web:SiteConfigResponse",
-                  "description": "Configuration of the app."
-              }
-          },
-          "type": "object"
+        "description": "A web app, a mobile app backend, or an API app.",
+        "properties": {
+          "siteConfig": {
+            "type": "object",
+            "$ref": "#/types/azure-native:web:SiteConfigResponse",
+            "description": "Configuration of the app."
+          }
+        },
+        "type": "object"
       }
     },
     "azure-native:alertsmanagement:getActionRuleByName": {

